### PR TITLE
Basic Packagedcode module for handling Rust's cargo crates

### PR DESCRIPTION
**Pull request by ritiek**
A very basic implementation at the moment. This is still missing tests which I'll add in shortly.

Otherwise, this is ready for review!

### CODE REVIEW

**Comment by ritiek** [comment]
Also see #1424.

**Comment by ritiek** [comment]
For those looking to quickly test this out. Clone any Rust repository and run scancode on it:
```
$ git clone https://github.com/jvns/snake.rs
$ scancode --package snake.rs/ --json-pp -
```

**Comment by ritiek** [comment]
I've added some tests as well.

This should be ready to merge if everything looks good. I think we could extend the subclass for handling more `Cargo.toml` attributes in an another PR. As it would be a big task in itself.

File: tests/packagedcode/data/rubygems/gemspec/cat.gemspec.expected.json [hunk 1]
```diff
@@ -32785,6 +32785,9 @@
     "thirdparty/text_unidecode-1.2-py2.py3-none-any.whl", 
     "thirdparty/text_unidecode-1.2-py2.py3-none-any.whl.ABOUT", 
     "thirdparty/text_unidecode-1.2-py2.py3-none-any.whl.NOTICE", 
+    "thirdparty/toml-0.10.0-py2.py3-none-any.whl",
```
Comment (pombredanne) [inline tests/packagedcode/data/rubygems/gemspec/cat.gemspec.expected.json]: That's not your making but this is a weird test that I will need to fix.

**Pull request merged**